### DISCUSSION TITLE
Running into error: No template for dependency: ConstDependency

### DISCUSSION THREAD

**Discussion by liuliangsir** [privatenumber/webpack-localize-assets-plugin#44] [Q&A] [answered]
## Bug description
webpack throws the error above after building `webpack-localize-asets-plugin` and linking the generated asset to some project

## Expected behavior
No error

## Reproduction steps
- Clone `webpack-localize-asets-plugin` to local machine
- Run npm script `build`
- Link the generated asset to some project which is going to use `webpack-localize-asets-plugin`
- Launch that project with some npm scripts

## Environment

- webpack-localize-asets-plugin version: 0.0.0-semantic-release
- Webpack version: 4.44.2
- Operating System: 

  ![image](https://user-images.githubusercontent.com/8049878/149656852-5037e92d-419e-4652-9468-7fd506459a9f.png)

- Node version: v14.16.0
- Package manager (npm/yarn/pnpm) and version: pnpm 6.26.1

**Comment by privatenumber** [comment]
Do you encounter this problem if you directly install from npm?

**Comment by liuliangsir** [comment]
@privatenumber no the error above with npm install

**Comment by privatenumber** [comment]
This is not a bug then. Sounds like a problem in how you're setting up your dev environment (eg. with npm link).

Moving this to Discussions.

**Comment by privatenumber** [comment, marked as answer]
You're probably getting this error because when you `npm link`, requests from within `webpack-localize-assets-plugin` resolve its dependencies within its own `node_modules` ,which includes dev dependencies and a different version of Webpack from your application.

This is a debugging flaw in npm (link, specifically). Instead, try `npm install <directory>` to do a clean install of the local package without dev dependencies.

**Comment by liuliangsir** [comment]
@privatenumber BTW, how to use `ConstDependency` to replace a string like __('Hello' + foo + 'world') with `'你好，世界' + foo`

**Reply by privatenumber** [reply]
I don't really understand what you're asking specifically, but you can look at how it's being used in the code:
https://github.com/privatenumber/webpack-localize-assets-plugin/blob/793b24fea2651c1f37807bac15d531813a2ad402/src/index.ts#L198

**Comment by sebastianramirez-tributi** [comment]
I'm getting the same issue when I run my app with 
webpack@5.74.0
webpack-localize-asets-plugin@1.5.2
The config I have it's
```js
  plugins: [
    ...
    new LocalizeAssetsPlugin({
      locales: {
        es: path.resolve(__dirname, './locales/es.json'),
      },
      functionName: 'fnTranslation',
    }),
  ]
```

and in the component is using like this
```jsx
     <span>{fnTranslation('header.title')}</span>
```

when I run the app in development mode or run the build command the error it's the same, not sure If I'm missing something
I already did a clean install
please help!
Thanks

**Reply by privatenumber** [reply]
If you really want help, I suggest providing a _minimal_ reproduction repository to make it easy for others to debug.
The information you provided here isn't sufficient for others to help.

**Reply by sebastianramirez-tributi** [reply]
Finally, I could get a fix for that. To be on the same page, I have a monorepo with root `package.json` and also for each app. It turns out I installed the library only in apps scope and I had that error, but when I moved to root and finally could run properly.
